src/ts/ssh-tcp/services/portForwardingService.ts

```diff
@@ -749,15 +749,20 @@ export class PortForwardingService extends SshService {
 		forwardedPort: number,
 		cancellation?: CancellationToken,
 	): Promise<boolean> {
-		const forwarder = this.localForwarders.get(forwardedPort);
-		if (!forwarder) {
-			return false;
-		}
-
-		this.localForwarders.delete(forwardedPort);
-		forwarder.dispose();
+		if (this.acceptLocalConnectionsForForwardedPorts) {
```

Comment:
[acceptLocalConnectionsForForwardedPorts](https://github.com/microsoft/dev-tunnels-ssh/blob/7623601710eb74327f05dffbdcfa2f7399d4cb56/src/ts/ssh-tcp/services/portForwardingService.ts#L112) may be changed during the session:
> This property is typically initialized before connecting a session (if not keeping the
>	default). It may be changed at any time while the session is connected, and the new value
>	 will affect any newly forwarded ports after that, but not previously-forwarded ports.

If this happens, it may have unintentional consequences in `cancalForwarding`. 

I think the fix may be resilient though:
``` ts
		const forwarder = this.localForwarders.get(forwardedPort);
		if (forwarder) {
    		this.localForwarders.delete(forwardedPort);
	   	forwarder.dispose();
         return true;
		}
     
    const port = new ForwardedPort(forwardedPort, forwardedPort, true);
    if (this.remoteForwardedPorts.has(port)) {
    		this.remoteForwardedPorts.removePort(port);
			return true;
    }
    
    return false;
```